pennylane/qnode.py

```diff
@@ -1024,35 +1066,21 @@ def _pd_analytic(self, params, idx, force_order2=False, **kwargs):
                 B = np.eye(1 +2*w)
                 B_inv = B.copy()
                 for BB in self._op_successors(o_idx, 'G'):
-                    temp = BB.heisenberg_tr(w)
-                    B = temp @ B
-                    temp = BB.heisenberg_tr(w, inverse=True)
-                    B_inv = B_inv @ temp
+                    if not BB.supports_heisenberg:
+                        continue
```

Comment:
:+1: Might be worth adding a comment though
```suggestion
                        # if the successor gate is non-Gaussian in analytic differentiation
                        # mode, then there must be no observable following it.
                        continue
```